File: src/main/kotlin/leetCode/medium/ContinuousSubArraySum.kt
package leetCode.medium

class ContinuousSubArraySum {
    fun checkSubarraySum(nums: IntArray, k: Int): Boolean {
        val prefixSum = IntArray(nums.size)

        var sum = 0
        for (i in nums.indices) {
            sum += nums[i]
            prefixSum[i] = sum
        }

        var memory = mutableMapOf<Int, Int>()

//        for (i in 0 until nums.size) {
//
//            if (memory[])
//        }

        return false
    }
}

File: go/leetCode/medium/continuous_subarray_sum/continuous_subarray_sum.go
package continuous_subarray_sum

func checkSubarraySum(nums []int, k int) bool {
	tmp := make(map[int]bool, 0)

	prefixSum := 0
	for i, num := range nums {
		tmp[num] = true
		prefixSum += num

		remainder := prefixSum % k
		if remainder == 0 || tmp[remainder] || num == 0 && i >= 1 {
			return true
		}
	}

	return false
}
